python/graphstorm/eval/eval_func.py

```diff
@@ -103,6 +103,12 @@ def __init__(self):
         self.metric_function["mse"] = compute_mse
         self.metric_function["mae"] = compute_mae
 
+        # This is the operator used to measure each metric performance in evaluation
+        self.metric_eval_function = {}
```

Comment:
I added this to make all Evaluator following the same behavior in the `compute_score()` method. 

Before the PR, only Classification Evaluator support using different metrics for `train=True/False`. Now all Evaluators support this behavior. In addition, because the `GSgnnPredictionEvalInterface`'s `compute_score(self, pred, labels, train=True)` set `train` argument. We need to use it in all Evalutors to avoid the `unused argument` pylint error.